### ISSUE TITLE
Windows: new_local_repository doesn't get re-fetched after file changing

### ISSUE THREAD

**Issue by meteorcloudy** [bazelbuild/bazel#3374]
A simple reproduce:
```
$ tree .
.
├── A
│   └── bin.py
└── B
    ├── BUILD # empty
    └── WORKSPACE

$ cat A/bin.py
print("Hello World")

$ cat B/WORKSPACE
new_local_repository(
  name = "A",
  path = "C:/tools/msys64/home/pcloudy/workspace/my_tests/new_local_repository_test/A",
  build_file_content = """
py_binary(
  name = "bin",
  srcs = ["bin.py"],
)
  """,
)

$ cd B/

$ bazel build @A//:bin
INFO: Analysed target @A//:bin.
INFO: Found 1 target...
Target @A//:bin up-to-date:
  C:/tmp/_bazel_pcloudy/us9pue2o/execroot/__main__/bazel-out/msvc_x64-fastbuild/bin/external/A/bin
INFO: Elapsed time: 1.497s, Critical Path: 0.31s
INFO: Build completed successfully, 7 total actions

$ ./bazel-bin/external/A/bin
Hello World

$ echo "print(\"Hi\")" > ../A/bin.py

$ bazel build @A//:bin
INFO: Analysed target @A//:bin.
INFO: Found 1 target...
Target @A//:bin up-to-date:
  C:/tmp/_bazel_pcloudy/us9pue2o/execroot/__main__/bazel-out/msvc_x64-fastbuild/bin/external/A/bin
INFO: Elapsed time: 0.433s, Critical Path: 0.01s
INFO: Build completed successfully, 1 total action

$ ./bazel-bin/external/A/bin    # Should print "Hi"
Hello World    

```

As far as I know, local repository should be re-fetched every time.

A few things I noticed:

- `bazel clean` will trigger a re-fetch
- `bazel shutdown` will trigger a re-fetch
- This only happens on Windows.
- A bug in both 0.5.2 and HEAD

**Comment by meteorcloudy** [comment]
FYI, @kchodorow

**Comment by meteorcloudy** [comment]
I believe the root cause is again because symlinking on file is copy on Windows.
See [here](https://github.com/bazelbuild/bazel/blob/1e0628d18c2e3defaddd935ffb41edbf3ef40339/src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryFunction.java#L435)
```
  public static boolean symlinkLocalRepositoryContents(
      Path repositoryDirectory, Path targetDirectory)
      throws RepositoryFunctionException {
    try {
      FileSystemUtils.createDirectoryAndParents(repositoryDirectory);
      for (Path target : targetDirectory.getDirectoryEntries()) {
        Path symlinkPath = repositoryDirectory.getRelative(target.getBaseName());
        createSymbolicLink(symlinkPath, target);
      }
    } catch (IOException e) {
      throw new RepositoryFunctionException(e, Transience.TRANSIENT);
    }

    return true;
  }
```

If `target` is a file, it would be a copy.

But I don't understand fully yet is when will a local repository get re-fetched. It doesn't check the marker data AFAIK.

**Comment by laszlocsomor** [comment]
Is this bug fixed?

**Comment by laszlocsomor** [comment]
This could be related to https://github.com/bazelbuild/bazel/issues/6351

**Comment by laszlocsomor** [comment]
This is fixed in Bazel 0.23.1
